packages/core/src/types/arazzo.ts

```diff
@@ -1,391 +1,308 @@
-import { getNodeTypesFromJSONSchema } from './json-schema-adapter';
+import { mapOf, type NodeType, listOf } from '.';
+import { DependentRequired, Schema, SchemaProperties } from './oas3_1';
+import { Discriminator, DiscriminatorMapping, ExternalDocs, Xml } from './oas3';
 
-import type { NodeType } from '.';
-
-export const ARAZZO_ROOT_TYPE = 'Root';
-
-export const operationMethod = {
-  type: 'string',
-  enum: ['get', 'post', 'put', 'delete', 'patch'],
-} as const;
-export const expectSchema = {
-  type: 'object',
+const Root: NodeType = {
   properties: {
-    statusCode: { type: 'number' },
-    mimeType: { type: 'string' },
-    body: {},
-    schema: {
+    arazzo: { type: 'string', enum: ['1.0.0'] },
+    info: 'Info',
+    sourceDescriptions: 'SourceDescriptions',
+    'x-parameters': 'Parameters',
+    workflows: 'Workflows',
+    components: 'Components',
+  },
+  required: ['arazzo', 'info', 'sourceDescriptions', 'workflows'],
+  extensionsPrefix: 'x-',
+};
+const Components: NodeType = {
+  properties: {
+    inputs: 'NamedInputs',
+    parameters: {
+      type: 'object',
+      additionalProperties: 'ParameterObject',
+    },
```

Comment:
Let's have separate names for components (and others too):

```suggestion
    parameters: 'NamedParameters',
    },
```